regression/test.pl

```diff
@@ -65,7 +65,7 @@ ($$)
 
   chomp @data;
   if($exit_signal_checks) {
-    grep { /^\^EXIT=[\(\|\d]*\d+\)?\$$/ } @data or die "$fname: Missing EXIT test\n";
+    grep { /^\^?EXIT=[\(\|\d]*\d+\)?\$$/ } @data or die "$fname: Missing EXIT test\n";
```

Comment:
I'm aware this is work-in-progress, but I'd really like to get a better understanding of what's going on here. The `EXIT=...` output is generated by `test.pl` itself. If that isn't at the beginning of a line, then something else has gone wrong.

Reply:
Still experimenting with that. I have some tests that succeed when they are killed by a timeout. In that case the EXIT output can get appended to whatever line is being printed at that moment. Surprisingly, printing `\nEXIT...` in test.pl doesn't put it on a new line either...

Reply:
Is that a race condition with some buffers only being flushed? You might try doing
````perl
$| = 1;
print "\n";
````
before it's printing `EXIT`.

Reply:
Just noting: this change no longer appears in #5217